src/App.tsx

```diff
@@ -1,18 +1,70 @@
-import React from 'react';
+import React, { useEffect } from 'react';
 import './App.css';
-import { getNumbers } from './utils';
+import { useSearchParams } from 'react-router-dom';
+import { getNumbers, getLastPage } from './utils';
+import { Pagination } from './components/Pagination';
 
-// eslint-disable-next-line @typescript-eslint/no-unused-vars
-const items = getNumbers(1, 42)
-  .map(n => `Item ${n}`);
+const ITEMS = getNumbers(1, 42).map(n => `Item ${n}`);
+const PER_PAGE_VALUES = [3, 5, 10, 20];
+const DEFAULT_PARAMS = {
+  page: '1',
+  perPage: '5',
+};
 
 export const App: React.FC = () => {
+  const [searchParams, setSearchParams] = useSearchParams();
+
+  const total = ITEMS.length;
+  const currentPage = Number(searchParams.get('page') || DEFAULT_PARAMS.page);
+  const perPage = Number(searchParams.get('perPage') || DEFAULT_PARAMS.perPage);
+  const lastPage = getLastPage(total, perPage);
+
+  const fromIndex = currentPage * perPage - perPage;
+  const toIndex = currentPage === lastPage ? total : currentPage * perPage;
+  const ITEMSOnPage = ITEMS.slice(fromIndex, toIndex);
```

Comment:
btw, it's not the constant, 'cause this var depends on some other vars, that are not the constants either)